src/parsing/astHelper.ts

```diff
@@ -190,11 +190,28 @@ function getAttribute(ast: IAST, attributeName: string): string | null {
 function getQName(ast: IAST): QName {
 	return {
 		localName: getTextContent(ast),
-		namespaceURI: getAttribute(ast, 'URI'),
-		prefix: getAttribute(ast, 'prefix') || '',
+		namespaceURI: getAttribute(ast, 'URI') as string,
+		prefix: (getAttribute(ast, 'prefix') as string) || '',
 	};
 }
 
+/**
+ * Insert an attribute with a given name and value to the AST.
+ *
+ * @param ast The ast node to insert the attribute to
+ * @param name The name of the attribute
+ * @param data The data of the attribute
+ */
+function insertAttribute(ast: IAST, name: string, data: any) {
```

Comment:
data any? Surely you mean `string|SequenceType`?

Reply:
To my knowledge this is meant to be a very generic function that can really be used to insert anything into the AST.